src/a-runtime-console/kubernetes/store/oauth-config-store.spec.ts

```diff
@@ -104,26 +90,24 @@ describe('OauthConfigStore', () => {
             // provide OAuthConfigStore with a factory inside the fakeAsync zone
             // so the mockBackend can catch http requests made inside the constructor
             provide: OAuthConfigStore, useFactory: fakeAsync((
-              http: Http,
-              mockBackend: MockBackend,
+              http: HttpClient,
+              controller: HttpTestingController,
               logger: Logger,
               errorHandler: ErrorHandler,
               notifications: NotificationsService
             ) => {
-              mockBackend.connections.subscribe((connection: MockConnection) => {
-                connection.mockRespond(new Response(new ResponseOptions({
-                  body: JSON.stringify(data),
-                  status: 200
-                })));
-              });
+              const req = controller.expectOne('/_config/oauth.json');
+              expect(req.request.method).toBe('GET');
```

Comment:
This approach did work for the first describe block but somehow its not working for other describe blocks. If i `xdescribe` the first block the second one passes with third and fourth failing. It seems some value is not resetting with each describe block which is why if first runs with certain values other ones fail because of same values.

Reply:
Try merging them into one describe block with different `req.flush(<data>)` calls. The test runner has some issues with multiple calls to `TestBed.configureTestingModule`.